vector/src/main/java/im/vector/app/features/home/room/detail/upgrade/UpgradeRoomViewModelTask.kt

```diff
@@ -84,8 +84,6 @@ class UpgradeRoomViewModelTask @Inject constructor(
 //                                autoJoin = currentInfo.autoJoin ?: false,
                                 suggested = currentInfo.suggested
                         )
-
-                        parentSpace.removeChildren(params.roomId)
```

Comment:
I would guess this was the first attempt at removing the duplication, and though it worked for that purpose, you would no longer see the pre-upgraded room in the space but you'd see it in Home

Reply:
This could be valuable for other cases, some counters could use spaces children number or some check (if space is empty or not), so absence of removal could cause some side effects.

@BillCarsonFr you added this line, maybe you could verify if removing it could cause any problems?

Reply:
I was thinking about some places, where we could get rooms without applying this filter, like getting space''s room count or something like that. 🤔